src/hyperloglog.c

```diff
@@ -1041,6 +1045,120 @@ int hllAdd(robj *o, unsigned char *ele, size_t elesize) {
     }
 }
 
+#if defined(HAVE_AVX2)
+/* Optimized for the default configuration: 16384 registers of 6 bits each. */
+ATTRIBUTE_TARGET_AVX2
+void hllMergeDense_AVX2(uint8_t *reg_raw, const uint8_t *reg_dense) {
+    const __m256i shuffle = _mm256_setr_epi8( //
+        4, 5, 6, -1,                             //
+        7, 8, 9, -1,                             //
+        10, 11, 12, -1,                          //
+        13, 14, 15, -1,                          //
+        0, 1, 2, -1,                             //
+        3, 4, 5, -1,                             //
+        6, 7, 8, -1,                             //
+        9, 10, 11, -1                            //
+    );
+
+    /* Merge the first 32 registers normally 
+     * as the AVX2 algorithm needs 4 padding bytes at the start */
+    uint8_t val;
+    for (int i = 0; i < 32; i++) {
```

Comment:
```suggestion
    for (int i = 0; i < 8; i++) {
```
as you comments said, it just needs to 4 padding bytes, so here just skip 8 bytes, also is aligned, of course, we also update the blow

Reply:
changed in b08a2384830fb44af49364be1ac514b9345c05ba